doc/style_guide.md

```diff
@@ -1,151 +0,0 @@
-# Developer Guide
```

Comment:
This document should not be removed in my opinion. There is a lot of useful information here.

Reply:
This moved to CONTRIBUTING.md

Reply:
Seems like we are loosing a lot of important information in the process.

Reply:
No, this entire style guide document has been replaced and is irrelevant now.

Reply:
The problem with trying to maintain a `.stylelintrc`/`.eslintrc` and a separate description of those rules is that there are then two sources of truth and the two will never stay in sync. It's best to have people refer to once source of truth, which is the place where the rules are configured, `.stylelintrc` and `.eslintrc`.

Reply:
True, updating two locations for one change doesn't make sense. I still feel like we are loosing something by not having real examples but if I am in the minority on this then I'm not going to hold up the PR over it.